packages/cli/src/lock.rs

```diff
@@ -0,0 +1,200 @@
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+use base64::Engine;
+use serde::{Deserialize, Serialize};
+
+use crate::plugin::CliPlugin;
+
+#[derive(Default, Debug, Clone, Deserialize, Serialize)]
+pub struct DioxusLock {
+    #[serde(skip)]
+    pub path: PathBuf,
+    pub plugins: HashMap<String, PluginLockState>,
+}
+
+impl DioxusLock {
+    pub fn new(path: PathBuf) -> Self {
+        Self {
+            path,
+            plugins: HashMap::new(),
+        }
+    }
+
+    pub fn load() -> crate::error::Result<Self> {
+        let crate_dir = crate::cargo::crate_root()?;
+
+        let crate_dir = crate_dir.as_path();
+
+        let Some(dioxus_conf_file) = acquire_dioxus_lock(crate_dir) else {
+            return Ok(Self::new(crate_dir.join("Dioxus.lock")));
```

Comment:
It might be better to output Dioxus.lock inside the target directory or the new `.dioxus` directory so that people don't have to add it to their `.gitignore`. It shouldn't be something that users need to know about